pkg/gcv/cf/cf.go

```diff
@@ -81,41 +78,49 @@ func (cf *ConstraintFramework) validateTemplate(t *configs.ConstraintTemplate) e
 	return err
 }
 
-// AddTemplate tracks an additional constraint template. This template is only used if a constraint is provided.
-func (cf *ConstraintFramework) AddTemplate(template *configs.ConstraintTemplate) error {
-	if _, exists := cf.templates[template.GeneratedKind]; exists {
-		return status.Errorf(codes.AlreadyExists, "Conflicting constraint templates with kind %s from file %s", template.GeneratedKind, template.Confg.FilePath)
-	}
-	if err := cf.validateTemplate(template); err != nil {
-		return status.Error(codes.InvalidArgument, err.Error())
+// Configure will set the constraint templates and constraints for ConstraintFramework
+func (cf *ConstraintFramework) Configure(templates []*configs.ConstraintTemplate, constraints []*configs.Constraint) error {
+	// create compiler from templates, other rego sources
+	templateMap := make(map[string]*configs.ConstraintTemplate)
+	for _, template := range templates {
+		if _, exists := templateMap[template.GeneratedKind]; exists {
+			return errors.Errorf("conflicting constraint templates with kind %s from file %s", template.GeneratedKind, template.Confg.FilePath)
```

Comment:
Shouldn't this continue to return the `codes.AlreadyExists` error (like constraints do)?

Reply:
Not necessary since this is a 'set' on all the templates.  Also, we don't programmatically handle those errors anywhere (they aren't returned in any RPC calls or handled on startup), so there's no need to return RPC error codes here.  Also, when we switch to CF, CF won't be returning gRPC error codes.

Reply:
We should still be consistent with constraints though, IMO: https://github.com/forseti-security/config-validator/pull/60/files/f84b630ae495bb83317b1de92043cf8cd082cccb#diff-9d6e91de36b00c7136016437773bd049R102

Reply:
I'll update this in the morning.  Keep in mind that pkg/gcv/cf is going to get deleted once we integrate with the real constraint framework.

Reply:
Ok. Once that's updated, this is good to merge.

Reply:
Ok, I've removed gRPC related status codes from validator.go / cf.go